sdk/src/azure/platform/az_mosquitto5.c

```diff
@@ -0,0 +1,591 @@
+
+  if (me->_internal.options.disable_tls == 0)
+  {
+    bool use_os_certs
+        = (az_span_ptr(me->_internal.options.certificate_authority_trusted_roots) == NULL);
+
+    if (use_os_certs)
+    {
+      _az_RETURN_IF_FAILED(_az_result_from_mosq(
+          mosquitto_int_option(me->_internal.mosquitto_handle, MOSQ_OPT_TLS_USE_OS_CERTS, 1)));
+    }
+
+    _az_RETURN_IF_FAILED(_az_result_from_mosq(mosquitto_tls_set(
+        me->_internal.mosquitto_handle,
+        (const char*)az_span_ptr(me->_internal.options.certificate_authority_trusted_roots),
+        use_os_certs ? REQUIRED_TLS_SET_CERT_PATH : NULL,
+        (const char*)az_span_ptr(connect_data->certificate.cert),
+        (const char*)az_span_ptr(connect_data->certificate.key),
+        NULL)));
+  }
+
+  if (az_span_ptr(connect_data->username) != NULL && az_span_ptr(connect_data->password) != NULL)
+  {
+    _az_RETURN_IF_FAILED(_az_result_from_mosq(mosquitto_username_pw_set(
+        me->_internal.mosquitto_handle,
+        (const char*)az_span_ptr(connect_data->username),
+        (const char*)az_span_ptr(connect_data->password))));
+  }
+
+  _az_RETURN_IF_FAILED(_az_result_from_mosq(mosquitto_connect_async(
+      (struct mosquitto*)me->_internal.mosquitto_handle,
+      (char*)az_span_ptr(connect_data->host),
+      connect_data->port,
+      AZ_MQTT5_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS)));
+
+  _az_RETURN_IF_FAILED(_az_result_from_mosq(mosquitto_loop_start(me->_internal.mosquitto_handle)));
+
+  return ret;
+}
+
+AZ_NODISCARD az_result az_mqtt5_outbound_sub(az_mqtt5* mqtt5, az_mqtt5_sub_data* sub_data)
+{
+  return _az_result_from_mosq(mosquitto_subscribe_v5(
+      mqtt5->_internal.mosquitto_handle,
+      &sub_data->out_id,
+      (char*)az_span_ptr(sub_data->topic_filter),
+      sub_data->qos,
+      0,
+      sub_data->properties ? sub_data->properties->_internal.options.properties : NULL));
+}
+
+AZ_NODISCARD az_result az_mqtt5_outbound_pub(az_mqtt5* mqtt5, az_mqtt5_pub_data* pub_data)
+{
+  return _az_result_from_mosq(mosquitto_publish_v5(
+      mqtt5->_internal.mosquitto_handle,
+      &pub_data->out_id,
+      (char*)az_span_ptr(pub_data->topic), // Assumes properly formed NULL terminated string.
+      az_span_size(pub_data->payload),
+      az_span_ptr(pub_data->payload),
+      pub_data->qos,
+      false,
+      pub_data->properties ? pub_data->properties->_internal.options.properties : NULL));
+}
+
+AZ_NODISCARD az_result az_mqtt5_outbound_disconnect(az_mqtt5* mqtt5)
+{
+  return _az_result_from_mosq(mosquitto_disconnect_v5(mqtt5->_internal.mosquitto_handle, 0, NULL));
+}
+
+AZ_NODISCARD az_result az_mqtt5_property_bag_init(
+    az_mqtt5_property_bag* property_bag,
+    az_mqtt5* mqtt5,
+    az_mqtt5_property_bag_options const* options)
+{
+  (void)mqtt5;
+  property_bag->_internal.options
+      = options == NULL ? az_mqtt5_property_bag_options_default() : *options;
```

Comment:
We're keeping the pattern of initializing with options which involves setting, though for a mosquitto specific case it would make sense to just allow options to be `NULL`. @CIPop thoughts? The above line is now:
`property_bag->properties = options == NULL ? NULL : options->properties;`